src/main/java/org/jabref/logic/exporter/ModsExporter.java

```diff
@@ -76,267 +49,343 @@ public void export(final BibDatabaseContext databaseContext, final Path file, Li
-                    } else if (StandardField.JOURNAL.equals(field)) {
-                        addJournal(value, relatedItem);
+                        addIdentifier(writer, StandardField.PMID, value);
+                    } else if (StandardField.PAGES.equals(field)) {
+                        addPart(parts, value);
+                    } else if (StandardField.VOLUME.equals(field)) {
+                        addPart(parts, value);
+                    } else if (StandardField.ISSUE.equals(field)) {
+                        addPart(parts, value);
                     }
-
-                    addOriginInformation(field, value, originInfo);
+                    trackOriginInformation(originItems, field, value);
                 }
-                mods.getModsGroup().add(originInfo);
 
-                addRelatedAndOriginInfoToModsGroup(relatedItem, partDefinition, mods);
-                modsCollection.getMods().add(mods);
-            }
+                writeOriginInformation(writer, originItems, fieldMap);
 
-            JAXBElement<ModsCollectionDefinition> jaxbElement = new JAXBElement<>(
-                    new QName(MODS_NAMESPACE_URI, "modsCollection"), ModsCollectionDefinition.class, modsCollection);
+                // Write related items
+                writeRelatedInformation(writer, parts, fieldMap);
+                writer.writeEndElement(); // end mods
+            }
 
-            createMarshallerAndWriteToFile(file, jaxbElement);
-        } catch (JAXBException ex) {
+            // end element and close
+            closeWriter(writer);
+        } catch (
+                XMLStreamException |
+                FileNotFoundException ex) {
             throw new SaveException(ex);
         }
     }
 
-    private void createMarshallerAndWriteToFile(Path file, JAXBElement<ModsCollectionDefinition> jaxbElement)
-            throws JAXBException {
+    private IndentingXMLStreamWriter createWriter(final Path file) throws XMLStreamException, FileNotFoundException {
+        XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
+        FileOutputStream fOutputStream = new FileOutputStream(file.toFile());
+        IndentingXMLStreamWriter writer = new IndentingXMLStreamWriter(outputFactory.createXMLStreamWriter(fOutputStream));
+        writer.setIndentStep("    ");
+        writer.writeDTD("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n");
+
+        writer.writeStartElement("mods", "modsCollection", MODS_NAMESPACE_URI);
+        writer.writeNamespace("mods", MODS_NAMESPACE_URI);
+        writer.writeNamespace("ns2", "http://www.w3.org/1999/xlink");
+        writer.writeNamespace("xsi", "http://www.w3.org/2001/XMLSchema-instance");
+        writer.writeAttribute("xsi", "http://www.w3.org/2001/XMLSchema-instance", "schemaLocation", MODS_SCHEMA_LOCATION);
+        return writer;
+    }
+
+    private void closeWriter(XMLStreamWriter writer) throws XMLStreamException {
+        writer.writeCharacters("\n");
```

Comment:
I would not hard-code linebreaks. Isn't that something, what the streamwriter does?

![image](https://user-images.githubusercontent.com/1366654/235272571-346a93cb-3dde-4aa7-ab77-2c7e51e803a0.png)